### DISCUSSION TITLE
Issue with Tailwind CSS

### DISCUSSION THREAD

**Discussion by karakaiyuliia** [cypress-io/cypress#29816] [Questions and Help] [answered]
Cypress does not work with Tailwind CSS. I was following the setup in this [repo](https://github.com/cypress-io/cypress-component-testing-apps) and none of the suggested setups work for me, they all result in the below error when simply trying to mount a component. 
![Screenshot 2024-07-08 at 7 03 09 PM](https://github.com/cypress-io/cypress/assets/93935356/cc1f1e4a-fd7b-443e-8a74-860b770ae3ad)

I compiled my tailwind css using `tailwindcss -i src/tailwind.css -o src/index.css`
In my `component.js`:
```
import '../../src/index.css';

import { mount } from 'cypress/react'

Cypress.Commands.add('mount', mount)
```

In `cypress.config.js`:

```
const { defineConfig } = require('cypress');

module.exports = defineConfig({
    component: {
        supportFile: 'cypress/support/component.js',
        devServer: {
            framework: 'next',
            bundler: 'webpack',
        },
    },
})
```
And my test:

```
import React from 'react';
import MyComponent from "./page";

describe('mycomponent', () => {
    it('renders', () => {
        cy.mount(<MyComponent/>)
    })
})
```

I am using React, Next.js and Tailwind.

**Comment by barak-kovrr** [comment]
What worked for me is to add the following code to styles.ts:

```ts
// Prevent Next.js CSS insertion error
if (!document.querySelector('#__next_css__DO_NOT_USE__')) {
  const styleNode = document.createElement('style');
  styleNode.id = '__next_css__DO_NOT_USE__';
  document.head.appendChild(styleNode);
}

export {};
```

and then import this file in component.ts

**Comment by jennifer-shehane** [comment, marked as answer]
That’s Next.js’s CSS insertion error: Next uses style-loader to inject CSS at runtime and needs a marker element, `<div id="__next_css__DO_NOT_USE__">`, in the DOM. When it’s missing, mounting throws.
Cypress normally scaffolds this for you — it belongs in `cypress/support/component-index.html`, so yours is likely missing it. Add it to the <head>:

```html
<head>
  <!-- ...existing meta/title... -->
  <!-- Used by Next.js to inject CSS. -->
  <div id="__next_css__DO_NOT_USE__"></div>
</head>
```

@barak-kovrr’s styles.ts does the same thing in JS, so either works, but adding it to component-index.html is the cleaner fix.

One note: you’re mounting a page (./page). For Next.js pages we recommend E2E over component testing, since server logic (getServerSideProps, server components) won’t run here.
